### PR TITLE
Query: Aggregate MethodTranslators for Math type

### PR DESCRIPTION
**Pull request by smitpatel**
See https://github.com/aspnet/EntityFramework/pull/7639#issuecomment-280563183

### CODE REVIEW

**Comment by smitpatel** [comment]
@maumar @anpete

File: src/Microsoft.EntityFrameworkCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace Microsoft.EntityFrameworkCore.Query.ExpressionTranslators.Internal
+{
+    /// <summary>
+    ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+    ///     directly from your code. This API may change or be removed in future releases.
+    /// </summary>
+    public class SqlServerMathTranslator : IMethodCallTranslator
+    {
+
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs]: NIT: Run code cleanup

File: src/Microsoft.EntityFrameworkCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            { nameof(Math.Abs), "ABS" },
+            { nameof(Math.Ceiling), "CEILING" },
+            { nameof(Math.Floor), "FLOOR" },
+            { nameof(Math.Pow), "POWER" }
+
+        };
+
+        public Expression Translate(MethodCallExpression methodCallExpression)
+        {
+            var method = methodCallExpression.Method;
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs]: CheckNotNull

File: src/Microsoft.EntityFrameworkCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        private readonly Dictionary<string, string> _supportedMethodTranslations = new Dictionary<string, string>
+        {
+            { nameof(Math.Abs), "ABS" },
+            { nameof(Math.Ceiling), "CEILING" },
+            { nameof(Math.Floor), "FLOOR" },
+            { nameof(Math.Pow), "POWER" }
+
+        };
+
+        public Expression Translate(MethodCallExpression methodCallExpression)
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs]: virtual

File: src/Microsoft.EntityFrameworkCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            var method = methodCallExpression.Method;
+            if (method.DeclaringType == typeof(Math))
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs]: Add another branch for Round.

Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs]: And Truncate

**Comment by smitpatel** [comment]
Updated.

**Comment by ErikEJ** [comment]
providers beware?

**Comment by smitpatel** [comment]
@ErikEJ @maumar - Isn't this just SQL Server specific? Is "providers-beware" needed? It would be upto providers how they would want to organize provider specific translators.

**Comment by ErikEJ** [comment]
For consistency and readability, I always strive for the same structure as Core, I assume this would also be the case if the SQLCE provider was in the Core repo?

**Comment by roji** [comment]
Agree with @ErikEJ, especially since the math operators are likely to be very similar across database. And although it's not strictly necessary, keeping providers' translators in line with SqlServer's helps maintain things.

**Pull request merged**